platform/admin/core/admin-core-logviewer/src/main/webapp/js/actions/index.js

```diff
@@ -13,33 +13,78 @@
  *
  **/
 
+import { MD5 } from 'object-hash'
+
 // for applying a filter to the logs
-export function filter (filter) {
-  return {
-    type: 'FILTER_LOGS',
-    filter: filter
-  }
-}
+export const filter = (filter) => ({
+  type: 'FILTER_LOGS',
+  filter: filter
+})
 
 // for appending new logs to the stored log list
 // note: user will not lose context when scrolling
-export function append (entries) {
-  return {
-    type: 'APPEND_LOGS',
-    entries: entries
-  }
-}
+export const append = (entries) => ({
+  type: 'APPEND_LOGS',
+  entries: entries.map((entry) => ({
+    hash: entry.hash || MD5(entry),
+    ...entry
+  }))
+})
 
 // used for expanding the max amount of stored logs displayed on screen
-export function grow () {
-  return {
-    type: 'GROW_DISPLAY_SIZE'
+export const grow = () => ({
+  type: 'GROW_DISPLAY_SIZE'
+})
+
+// expands logs with the given hash
+export const expandEntry = (hash) => ({
+  type: 'CHANGE_EXPANDED_ENTRY',
+  hash: hash
+})
+
+// called at beginning of fetching
+// prevents slow networks from making duplicate calls to the backend
+export const beginFetching = () => ({
+  type: 'BEGIN_FETCHING'
+})
+
+// called at the end of fetching
+export const endFetching = () => ({
+  type: 'END_FETCHING'
+})
+
+// toggle polling to the backend for new logs
+export const togglePolling = () => ({
+  type: 'TOGGLE_POLLING'
+})
+
+// retrieves new logs from the backend and appends any new ones
+export const fetch = (getLogs) => (dispatch, getState) => {
+  console.log("1. check fetch: " + getState().isFetching)
+  if (getState().isPolling && !getState().isFetching) {
+    dispatch(beginFetching())
+    console.log("2. after setting fetch: " + getState().isFetching)
+    console.log("3. Making the call...")
+    getLogs((err, logs) => {
+      console.log("4. Finished the call...")
+      if (err) {
```

Comment:
Added a drop-down banner to display the error and auto-pauses the polling to prevent more errors. I'll add screenshots in a moment.